solang-parser/src/pt.rs

```diff
@@ -1258,6 +1273,8 @@ pub struct Parameter {
     pub storage: Option<StorageLocation>,
     /// The optional identifier.
     pub name: Option<Identifier>,
+    /// An optional annotation '@annotation'.
+    pub annotation: Option<Annotation>,
```

Comment:
The other fields are ordered by their appearance in the source code (exception: loc is always first). Maybe annotation can go first after loc?